android/app/src/main/kotlin/coop/polypoly/polypod/OnboardingActivity.kt

```diff
@@ -33,14 +45,51 @@ class OnboardingActivity : AppCompatActivity() {
                 R.id.headline_sub to R.string.onboarding_slide3_sub_headline,
                 R.id.body_text to R.string.onboarding_slide3_body_text,
             ),
+            mapOf(
+                R.id.headline_main to R.string.onboarding_slide4_headline,
+                R.id.headline_sub to R.string.onboarding_slide4_sub_headline,
+                R.id.body_text to R.string.onboarding_slide4_body_text,
+            ),
         )
 
+        if (!Preferences.isFirstRun(baseContext)) {
+            strings = listOf(strings[2])
```

Comment:
This logic is a bit hairy, but I presume if it's not the first run, only slide three is shown? I suppose saving the third slide as a variable could make this a bit less error prone (if we insert another slide or change their order e.g.)

Reply:
Refactored this.